subsys/net/ip/icmpv6.c

```diff
@@ -304,13 +304,18 @@ static enum net_verdict handle_echo_request(struct net_pkt *orig)
 
 	iface = net_pkt_iface(orig);
 
+	payload_len = ntohs(NET_IPV6_HDR(orig)->len) -
+		net_pkt_ipv6_ext_len(orig) - NET_ICMPH_LEN;
+	if (payload_len < NET_ICMPV6_UNUSED_LEN) {
```

Comment:
I'd say this whole chunk of code is not the clearest one. `payload_len` doesn't seem to be used elsewhere. And then, I'd say we should calculate `icmp_pkt_len`, and check that it's no smaller than `NET_ICMPH_LEN + NET_ICMPV6_UNUSED_LEN`. Not insisting though, it's just going thru those steps is what took me to verify that this patch looks correct.

Reply:
The payload_len was not used earlier for some reason. Now it holds the length of the ICMP messge, and if it is too small, then we can safely drop it. In ICMP protocol, the first 4 bytes are mandatory, the latter 4 (the NET_ICMPV6_UNUSED_LEN define) depends on what packet type is being sent. For echo-req, the latter four bytes contain the sequence and identifier numbers.

Reply:
As there can be user payload data in icmp message, the payload_len is kind of descriptive in this case.